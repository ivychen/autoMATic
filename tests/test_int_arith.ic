int main() {
	int i;
	int j;

	i = 0;
	j = 5;
	i = i + j;
	j = i + j;

	println(i); // 5
	println(j); // 10
	println(j - 1); // 9
	println((j - 1) * 5); // 45

	j = j - 5; // 5
	if (i == j) println("i == j"); // true

	i = i - 1; // 4

	if (i == j) println("i == j"); // false
	if (i != j) println("i != j"); // true
	if (i < j) println("i < j"); // true
	if (i <= j) println("i <= j"); // true
	if (j <= j) println("j <= j"); // true

	if (j <= i) println("j <= i"); // false
	if (j > i) println("j > i"); // true
	if (j >= i) println("j >= i"); // true
	if (j >= j) println("j >= j"); // true

        i = 8;
        j = 2;
        println(i / j); // 4
        println(i / (j * 2)); // 2
        println(j ** 3); // 8

	return 0;
}
