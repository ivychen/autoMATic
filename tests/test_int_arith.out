5
10
9
45
i == j
i != j
i < j
i <= j
j <= j
j > i
j >= i
j >= j
